src/spotify_to_tidal/sync.py

```diff
@@ -221,7 +265,10 @@ def sync_playlist(spotify_session: spotipy.Spotify, tidal_session: tidalapi.Sess
         print(f"No playlist found on Tidal corresponding to Spotify playlist: '{spotify_playlist['name']}', creating new playlist")
         tidal_playlist =  tidal_session.user.create_playlist(spotify_playlist['name'], spotify_playlist['description'])
     tidal_track_ids = []
-    spotify_tracks, cache_hits = TidalPlaylistCache(tidal_playlist).search(spotify_session, spotify_playlist)
+    spotify_tracks = get_tracks_from_spotify_playlist(spotify_session, spotify_playlist)
+    track_cache = TidalTrackCache()
+    track_cache.populate(spotify_tracks, tidal_playlist.tracks())
+    spotify_tracks, cache_hits = track_cache.search(spotify_session, spotify_playlist)
```

Comment:
This should be changed to `cache_hits = track_cache.search(spotify_tracks)`. Currently 
 `get_tracks_from_spotify_playlist()` is getting called twice

Reply:
Here's a patch which should work relative to #39:
[0001-Fix-Spotify-playlist-tracks-queried-twice.patch](https://github.com/spotify2tidal/spotify_to_tidal/files/15404911/0001-Fix-Spotify-playlist-tracks-queried-twice.patch)